src/uvc-sensor.h

```diff
@@ -43,6 +43,20 @@ class uvc_sensor : public raw_sensor_base
         return action( *_device );
     }
 
+    
+    void power_for_duration( std::chrono::steady_clock::duration timeout = std::chrono::milliseconds( 500 ) )
+    {
+        acquire_power();
+        std::weak_ptr< uvc_sensor > weak = std::dynamic_pointer_cast< uvc_sensor >( shared_from_this() );
+        std::thread release_power_thread( [weak, timeout]()
+        {
+            std::this_thread::sleep_for( timeout );
+            if( auto strong = weak.lock() )
+                strong->release_power();
```

Comment:
Please make sure `release_power();` cannot throw.
Even if today it can't might be better to wrap with try/catch as an exception here will crash the application